internal/executor/executor.go

```diff
@@ -260,7 +261,15 @@ func (executor *Executor) RunBuild(ctx context.Context) {
 
 		log.Printf("Executing %s...", command.Name)
 
-		stepResult, err := executor.performStep(subCtx, command)
+		var stepCtx context.Context
+
+		if command.ExecutionBehaviour == api.Command_ON_TIMEOUT {
+			stepCtx = context.Background()
```

Comment:
Should we do a 5 minutes timeout here? By default Cirrus sets a 15 deadline on Kubernetes pods but not on VMs. Seems having some timeout here is good safe guard.

Reply:
See https://github.com/cirruslabs/cirrus-ci-agent/pull/321/commits/62df77d1159a6078a37fca8aa6cdd3ab762aed49.